### ISSUE TITLE
Read ORC table failed

### ISSUE THREAD

**Issue by PengleiShi** [oap-project/gluten#575]
**Describe the bug**
Hi, i read an orc table with spark3.2.2 and gluten failed, the log is like below
![image](https://user-images.githubusercontent.com/23723660/202371180-c8ac0f82-4019-4f5b-9db1-69c7e4b0baa1.png)
![image](https://user-images.githubusercontent.com/23723660/202372909-1d3bc70d-fc81-4975-98e5-c0a9780f8bab.png)

And i execute `velox_example_scan_orc` which is one of velox examples, read orc files successfully. `velox_example_scan_orc` in OAP/velox also does successfully.

**To Reproduce**
```
create table if not exists test(
id bigint,
name string,
score int
) using orc;

insert into test values(1, "tom", 80), (2, "jerry", 80), (3, "jay", 80), (4, null, 90);

select * from test;
```

**Expected behavior**

**Additional context**

**Comment by rui-mo** [comment]
Spark generated ORC format is different with Velox's, and we previously made some conversions to generate the ORC file under Gluten to make it work. One big difference is about the RLE encoding. You may need use this config `--conf spark.hive.exec.orc.write.format=0.11` when generatign ORC in Spark to make it align with Velox's. Also, even with this config, previously there are some inconsitency on the int and bigint scanning, which may be fixed by Velox already. In Gluten, parquet file format is what we suggest and is tested frequently.

**Comment by FelixYBW** [comment]
several months ago, we noted velox uses dwrf format, not standard ORC format. Not sure their latest status.

**Comment by PengleiShi** [comment]
@rui-mo @FelixYBW  It looks like ORC reader has been supported in velox https://github.com/facebookincubator/velox/issues/858 , the code example given in velox, which reads an ORC file and then prints, can be executed properly with the spark generated ORC file. The same code example in OAP/velox also can be executed properly.

**Comment by FelixYBW** [comment]
@rui-mo Have we tested this PR? looks it's pretty old. It's already there when we create GLuten POC.

**Comment by rui-mo** [comment]
> @rui-mo Have we tested this PR? looks it's pretty old. It's already there when we create GLuten POC.

No, we did not test it. At that time, we already switched to dwrf format instead of previous converted ORC, see  [PR#189](https://github.com/oap-project/gluten/pull/189). ORC's support in Velox should be stronger than before since I noticed PRs to continously improve it.

Other PRs:
https://github.com/facebookincubator/velox/pull/2364
https://github.com/facebookincubator/velox/pull/2226

**Comment by rui-mo** [comment]
@PengleiShi I opened below PR to enable the usage for Velox's ORC reader. Would you like to verify if it can solve the issue you met?
https://github.com/oap-project/velox/pull/76

**Comment by PengleiShi** [comment]
> @PengleiShi I opened below PR to enable the usage for Velox's ORC reader. Would you like to verify if it can solve the issue you met? [oap-project/velox#76](https://github.com/oap-project/velox/pull/76)

Sure,  i will test it

**Comment by PengleiShi** [comment]
@rui-mo i have tested it on last friday, and there is a same error like above. Maybe it's my compile fault. And i will try to debug it.

**Comment by rui-mo** [comment]
> @rui-mo i have tested it on last friday, and there is a same error like above. Maybe it's my compile fault. And i will try to debug it.

Could you share with me the command you use to compile Gluten? Maybe the simplist way is to change the Velox repo and branch at below place and recompile Gluten, otherwise Velox should be manully compiled.

https://github.com/oap-project/gluten/blob/main/ep/build-velox/src/build_velox.sh#L17-L18

**Comment by PengleiShi** [comment]
```
mvn clean package -Pspark-3.2 -DskipTests -Dcheckstyle.skip -Dbuild_protobuf=OFF -Pbackends-velox -Dbuild_velox=ON -Dcompile_velox=ON -Dbuild_cpp=ON
```
I edited `/gluten_home/tools/build/velox_ep/velox/substrait/SubstraitToVeloxPlan.cpp ` manully.

**Comment by rui-mo** [comment]
> I also compiled gluten using this way https://github.com/oap-project/gluten/blob/main/ep/build-velox/src/build_velox.sh#L17-L18 , now the error message is below:**

Hi, @PengleiShi, this error means the orc reader is not registered. But I found dwrf reader was still used for orc scan and no orc reader was noticed. Maybe you could help check that. Below code shows how to register dwrf reader which we already done in Gluten.
```
dwrf::registerDwrfReaderFactory();
```
I also checked the unit test for Orc scan in Velox, and found it mocked some code to make the ORC scan work (https://github.com/facebookincubator/velox/blob/main/velox/dwio/dwrf/test/OrcTest.h#L71-L73).

**Comment by PengleiShi** [comment]
@weiting-chen hi, it seems orc read is not be supported yet, right?
![image](https://github.com/oap-project/gluten/assets/23723660/02069e53-8514-42c5-8c12-1195d2026982)
whether we should fallback to vanilla spark `FileSourceScanExec`  instead of native read?
![image](https://github.com/oap-project/gluten/assets/23723660/a1049194-c8b0-491a-80a0-2b4e3e041412)

**Comment by rui-mo** [comment]
@PengleiShi Since https://github.com/oap-project/gluten/pull/3341, ORC's support has been missing. Could you open a PR to fallback it in Gluten?

**Comment by PengleiShi** [comment]
@rui-mo  sry for the late reply.  https://github.com/oap-project/velox/pull/417 and https://github.com/oap-project/gluten/pull/3437 both are ongoing, so is it necessary to open a PR to fallback it?  Currently,  `spark.gluten.sql.columnar.filescan=false` is worked for me.

**Comment by rui-mo** [comment]
@PengleiShi If disabling scan through config is enough for you, maybe it's all right to continue waiting for the existing work on ORC.
